doc/whats-new.rst

```diff
@@ -63,6 +63,9 @@ Bug fixes
   By `Martin Pletcher <https://github.com/pletchm>`_.
 - Removed usages of `pytest.config`, which is deprecated (:issue:`2988`:)
   By `Maximilian Roos <https://github.com/max-sixty>`_.
+- Fixed a bug in `utils.safe_cast_to_index` and performance issues when
+  `cftime` is installed (:issue:`3000`:)
```

Comment:
nit: I think it might be helpful to be a little more specific here (also the trailing colon for the issue link is not needed -- looks like we may have missed that in the entry above)
```suggestion
  using :py:meth:`~xarray.DataArray.stack` when cftime is installed (:issue:`3000`)
```

Reply:
Actually `safe_cast_to_index` is called in from multiple files (alignment, groupby, indexing, utils), I'm not sure only `stack` was impacted

Reply:
Good point -- maybe let's say something like "performance issues when using a MultiIndex with cftime installed."  I'm reasonably confident this is MultiIndex-centric.

Reply:
> I'm reasonably confident this is MultiIndex-centric.

Performance will also be impacted for other object type indexes (like an index of strings), so ignore this too :).  @0x0L your suggested approach, https://github.com/pydata/xarray/pull/3001#issuecomment-499598244, fixes this too.